### PR TITLE
TAS: Add documentation page for concepts

### PR DESCRIPTION
**Pull request by mimowo**
#### What type of PR is this?

/kind documentation

#### Which issue(s) this PR fixes:

Part of #2724 

#### Special notes for your reviewer:

I think it would also be useful to have a tasks page, but I'm unlikely to be able to write it before 0.9.0.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: site/content/en/docs/concepts/topology_aware_scheduling.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+date: 2024-04-11
+weight: 6
+description: >
+  Allows scheduling of Pods based on the topology of nodes in a data center.
+---
+
+{{< feature-state state="alpha" for_version="v0.9" >}}
+
+It is common that AI/ML workloads require a significant amount of pod-to-pod
+communication, and thus the network bendwidth between the running Pods
```
Comment (dgrove-oss) [inline site/content/en/docs/concepts/topology_aware_scheduling.md:12]:
```suggestion
communication. Therefore the network bandwidth between the running Pods
```

File: site/content/en/docs/concepts/topology_aware_scheduling.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+translates into the workload execution time, and the cost of running
+such workloads. Then, the connectivity between the Pods depends on the placement
```
Comment (dgrove-oss) [inline site/content/en/docs/concepts/topology_aware_scheduling.md:14]:
```suggestion
such workloads. The available bandwidth between the Pods depends on the placement
```

File: site/content/en/docs/concepts/topology_aware_scheduling.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+within a rack, and there are multiple racks within a block. Pods running within
+the same organizational unit have better network bandwidth than Pods on
+different units. We say that nods placed in different racks are more distant
+than nodes placed within the same rack. Similarly, nodes placed in different
+blocks are more distant than two nodes within the same block.
+
+In this feature (called Topology Aware Scheduling, or TAS for short) we
+introduce a convention to represent the
+[hierarchical node topology information](#node-topology-information), and a set
+of APIs for Kueue administrators and users to utilize the information in order
```
Comment (dgrove-oss) [inline site/content/en/docs/concepts/topology_aware_scheduling.md:28]:
```suggestion
of APIs for Kueue administrators and users to utilize the information
```

File: site/content/en/docs/concepts/topology_aware_scheduling.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        nominalQuota: 100
+      - name: "memory"
+        nominalQuota: 100Gi
+```
+
+### User-facing APIs
+
+Once TAS is configured and ready to be used, you can create Jobs with the
+following annotations set at the PodTemplate level:
+- `kueue.x-k8s.io/podset-required-topology` - indicates that a PodSet requires
```
Comment (dgrove-oss) [inline site/content/en/docs/concepts/topology_aware_scheduling.md:105]:
```suggestion
- `kueue.x-k8s.io/podset-preferred-topology` - indicates that a PodSet requires
```

**Review by dgrove-oss — COMMENTED** [review]
A few minor suggestions.

File: site/content/en/docs/concepts/topology_aware_scheduling.md (hunk 4 above)
Comment (mimowo) [reply site/content/en/docs/concepts/topology_aware_scheduling.md]: thx

File: site/content/en/docs/concepts/topology_aware_scheduling.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+### Limitations
+
+Currently, there are multiple limitations for the compatibility of the feature
+with other features. In particular, a ClusterQueue referencing a TAS Resource
+Flavor (with the `.spec.topologyName` field) is marked as inactive in the
+following scenarios:
+- the CQ is in cohort
+- the CQ is using preemption
+- the CQ is using MultiKueue or ProvisioningRequest admission checks
```
Comment (tenzen-y) [inline site/content/en/docs/concepts/topology_aware_scheduling.md:159]:
```suggestion
- the CQ is using [MultiKueue](multikueue.md) or [ProvisioningRequest](../admission-check/controllers/provisioning.md) admission checks
```

File: site/content/en/docs/concepts/topology_aware_scheduling.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+As an admin, in order to enable the feature you need to:
+1. ensure the `TopologyAwareScheduling` feature gate is enabled
+2. create at least one instance of the `Topology` API
+3. reference the `Topology` API from a dedicated ResourceFlavor by the
+   `.spec.topologyName` field
+
+#### Example
+
+```yaml
```
Comment (tenzen-y) [inline site/content/en/docs/concepts/topology_aware_scheduling.md:64]: Could we move this to example folder?

https://github.com/kubernetes-sigs/kueue/tree/main/site/static/examples

File: site/content/en/docs/concepts/topology_aware_scheduling.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+	within the same topology domain corresponding to the topology level
+	indicated by the annotation value (e.g. within a rack or within a block).
+
+#### Example
+
+Here is an example Job a user might submit to use TAS. It assumes there exists
+a LocalQueue named `tas-user-queue` which refernces the ClusterQueue pointing
+to a TAS ResourceFlavor.
+
+```yaml
```
Comment (tenzen-y) [inline site/content/en/docs/concepts/topology_aware_scheduling.md:124]: ditto

File: site/content/en/docs/concepts/topology_aware_scheduling.md [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+We propose a lightweight model for representing the hierarchy of nodes within a
+data center by using node labels. In this model the node labels are set up by a
+cloud provider, or set up manually by administrators of on-premise clusters.
+
+Additionally, we assume that every node used for TAS has a set of the labels
+which identifies uniquely its location in the tree structure. We do not assume
+global uniqueness of labels on each level, i.e. there could be two nodes with
+the same "rack" label, but in different "blocks".
+
+For example, this is a representation of the dataset hierarchy;
```
Comment (tenzen-y) [inline site/content/en/docs/concepts/topology_aware_scheduling.md:42]:
```suggestion
For example, this is a representation of the data center hierarchy;
```

File: site/content/en/docs/concepts/topology_aware_scheduling.md [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+than nodes placed within the same rack. Similarly, nodes placed in different
+blocks are more distant than two nodes within the same block.
+
+In this feature (called Topology Aware Scheduling, or TAS for short) we
+introduce a convention to represent the
+[hierarchical node topology information](#node-topology-information), and a set
+of APIs for Kueue administrators and users to utilize the information
+to optimize the Pod placement.
+
+### Node topology information
```
Comment (tenzen-y) [inline site/content/en/docs/concepts/topology_aware_scheduling.md:31]: Could we mention that we only care about ready nodes?

File: site/content/en/docs/concepts/topology_aware_scheduling.md [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+      restartPolicy: Never
+```
+
+### Limitations
+
+Currently, there are multiple limitations for the compatibility of the feature
+with other features. In particular, a ClusterQueue referencing a TAS Resource
+Flavor (with the `.spec.topologyName` field) is marked as inactive in the
+following scenarios:
+- the CQ is in cohort
```
[the remark this answers is not in the document]
Comment (mimowo) [reply site/content/en/docs/concepts/topology_aware_scheduling.md]: done

File: site/content/en/docs/concepts/topology_aware_scheduling.md (hunk 5 above)
Comment (mimowo) [reply site/content/en/docs/concepts/topology_aware_scheduling.md]: done

File: site/content/en/docs/concepts/topology_aware_scheduling.md (hunk 6 above)
Comment (mimowo) [reply site/content/en/docs/concepts/topology_aware_scheduling.md]: done

File: site/content/en/docs/concepts/topology_aware_scheduling.md (hunk 9 above)
Comment (mimowo) [reply site/content/en/docs/concepts/topology_aware_scheduling.md:31]: done

File: site/content/en/docs/concepts/topology_aware_scheduling.md (hunk 8 above)
Comment (mimowo) [reply site/content/en/docs/concepts/topology_aware_scheduling.md]: done

**Review by tenzen-y — COMMENTED** [review]
Thank you!
/lgtm
/approve

**Comment by mimowo** [comment]
a race condition :) I have yet pushed a bit more comprehensive overview of the capacity calculation in  the last [update](https://github.com/kubernetes-sigs/kueue/compare/6f45058918d08bc8d87f734eba4251d18eb3b748..5cfa7ead0d79833056609c2c04af06f00238bb56). PTAL

**Review by tenzen-y — COMMENTED** [review]
Thanks.
/lgtm

**Pull request merged**